### ISSUE TITLE
[SC3045] `read -d` is POSIX.1-2024 compliant

### ISSUE THREAD

**Issue by JP-Ellis** [koalaman/shellcheck#3062]
## For bugs
- Rule Id: SC3045
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

## For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

## Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

read -r -d ':' foo
```

## Here's what shellcheck currently says:
```
Line 3	SC3045: In POSIX sh, read -d is undefined.
```
## Here's what I wanted or expected to see:

I believe this is a false-positive, as the POSIX description for `read` shows that it _does_ support `-d`. See the specification [here](https://pubs.opengroup.org/onlinepubs/9799919799/) and the relevant quote part below.

> ### NAME
> 
> `read` — read from standard input into shell variables
> 
> ### SYNOPSIS
> 
> ```
> read [-r] [-d delim] var...
> ```
> 
> ### OPTIONS
> 
> The read utility shall conform to XBD [12.2 Utility Syntax Guidelines](https://pubs.opengroup.org/onlinepubs/9799919799/basedefs/V1_chap12.html#tag_12_02) .
> 
> The following options shall be supported:
> 
> - `-d delim`
> 
>   If delim consists of one single-byte character, that byte shall be used as the logical line delimiter. If delim is the null string, the logical line delimiter shall be the null byte. Otherwise, the behavior is unspecified.
> 
> - `-r`
> 
>   Do not treat a <backslash> character in any special way. Consider each <backslash> to be part of the input line.

The support for `-d` in POSIX appears to be relatively recent though, as the previous edition do indeed omit it the option.

Perhaps I missed a way of specifying the POSIX version?

**Comment by tgross35** [comment]
This also has a pretty notable effect on [SC2044](https://www.shellcheck.net/wiki/SC2044), I am very much looking forward to having this flag.

Tested with Alpine and `ash` (busybox) supports `-d` already but unfortunately, Debian `dash` does not yet seem to support it yet. I requested via https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1087810.

(https://github.com/koalaman/shellcheck/issues/2555 is another issue relating to posix 2024)
